src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -45,11 +49,38 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         return serverList.stream().filter(server -> server.getId().equals(serverId)).findFirst();
     }
 
+    /**
+     * Returns a list of all servers that have been configured by the user. This can include incorrectly or illegally
+     * defined servers.
+     *
+     * @return
+     */
     public List<BitbucketServerConfiguration> getServerList() {
         return serverList;
     }
 
     public void setServerList(@Nonnull List<BitbucketServerConfiguration> serverList) {
```

Comment:
Not your code, but we don't need `@Nonnull` here since we have `ParametersAreNonnullByDefault` in the package-info

Reply:
I'm still a bit confused by this, there are a few references to `@Nonnull` in the code base. Should we be using it at all, and if so in what situation?

Reply:
We shouldn't be using `@Nonnull` at all because of the package-info. But it does mean that we have to remember to mark things as `@Nullable` instead.